internal/controllers/machine/machine_controller.go

```diff
@@ -712,6 +713,14 @@ func (r *Reconciler) shouldWaitForNodeVolumes(ctx context.Context, cluster *clus
 		return true, err
 	}
 
+	if noderefutil.IsNodeUnreachable(node) {
+		// If a node is unreachable, we can't detach the volume.
+		// We need to skip the detachment as we otherwise block deletions
+		// of unreachable nodes when a volume is attached.
+		log.Info("Skipping volume detachment as node is unreachable.")
```

Comment:
I think you meant false her?

Reply:
Woops, clear coffee deficiency. Indeed, got the logic the wrong way around. We should not wait for the detachment as the comment mentions.